src/main/java/org/opentripplanner/routing/api/request/request/filter/SelectRequest.java

```diff
@@ -20,7 +19,7 @@ public static Builder of() {
   private final List<MainAndSubMode> transportModes;
   private final AllowTransitModeFilter transportModeFilter;
   private final List<FeedScopedId> agencies;
-  private final RouteMatcher routes;
+  private final List<FeedScopedId> routes;
```

Comment:
What are the performance implications of using a List rather than a Set, which the original RouteMatcher used?

I think eventually we want to use the same indexing technique that unpreferred routes use, isn't it?

That's out of scope for this PR though.

Reply:
With a low number of entries (<~5), the List actually has better performance, since no hashCode needs to be calculated, and the `FeedScopedId` is easy to compare. We use List here for other things, I'll do some profiling and crate a PR if there is a difference.